### PR TITLE
Implemented onboarding flow progress indicator as per design

### PR DESCRIPTION
**Pull request by liamness**
## What does this change?

Fixes appearance. ~~Draft for now, checking I haven't broken anything in Chromatic.~~

## How to test

Testing via Storybook / Chromatic should be sufficient.

## How can we measure success?

✨

### Mobile

image

### Tablet

image

### Desktop

image

## Accessibility

- [ ] [Tested with screen reader](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#screen-readers)
- [ ] [Navigable with keyboard](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#keyboard-navigation)
- [ ] [Colour contrast passed](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#use-of-colour)

### CODE REVIEW

File: src/client/components/ConsentsSubHeader.tsx [hunk 1]
```diff
@@ -40,224 +48,177 @@ const h1ResponsiveText = css`
...
+export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
+  const active = current
+    ? (CONSENTS_PAGES_ARR as string[]).indexOf(current)
+    : 0;
+
+  const pageProgression = current && (
+    <div css={[progressWrapper, autoRow()]}>
+      <ol css={ol(active)}>
+        {CONSENTS_PAGES_ARR.map((page, index) => {
+          const status: PageStatus = (() => {
```
Comment (ob6160) [inline src/client/components/ConsentsSubHeader.tsx:188]: thoughts on splitting out this unnamed function into a named one?

File: src/client/components/ConsentsSubHeader.tsx [hunk 2]
```diff
@@ -40,224 +48,177 @@ const h1ResponsiveText = css`
...
+            if (index === active) {
+              return 'active';
+            }
+            if (index < active) {
+              return 'complete';
+            }
+            return 'pending';
+          })();
+          const screenReaderText = {
```
Comment (ob6160) [inline src/client/components/ConsentsSubHeader.tsx:197]: is there a case for adding `pending` to this list?

File: src/client/components/ConsentsSubHeader.tsx [hunk 3]
```diff
@@ -40,224 +48,177 @@ const h1ResponsiveText = css`
...
+            active: 'Current: ',
+            complete: 'Completed: ',
+          }[status as string];
+          return (
+            <li key={page} css={li(index, status)}>
+              <span>
+                {screenReaderText && (
+                  <span css={hiddenText}>{screenReaderText}</span>
```
Comment (ob6160) [inline src/client/components/ConsentsSubHeader.tsx:205]: great work with the screen reader consideration!

**Review by ob6160 — APPROVED** [review]
this is great work, it's so hard to avoid a "hacky" solution to implementing designs like this within time constraints and I think this is entirely fit for purpose, so I've avoided commenting on the bulk of the styling implementation–I'm happy if it renders alright :) great work with the accessibility affordance too! it would be a great extra to have a demo in the description see/hear how it'd appear to a user who uses a screen reader

File: src/client/components/ConsentsSubHeader.tsx [hunk 4]
```diff
@@ -40,224 +48,177 @@ const h1ResponsiveText = css`
...
+
+const progressWrapper = css`
+  padding: 0 ${CIRCLE_RADIUS}px;
+  overflow: hidden;
 `;
 
-const maxContentOverride = (isLastPage: boolean) => css`
-  width: ${isLastPage ? 'max-content' : 'initial'};
+const hiddenText = css`
+  display: none;
```
Comment (ob6160) [inline src/client/components/ConsentsSubHeader.tsx:176]: just leaving as a note here – as discussed, this should probably be `visibility: hidden;` / using the Source visually hidden rules

File: src/client/components/ConsentsSubHeader.tsx (hunk 2 above)
Comment (liamness) [reply src/client/components/ConsentsSubHeader.tsx:198]: I was working from this example: https://www.w3.org/WAI/tutorials/forms/multi-page/

I think it should be implied from the order of the list items and the status of the preceding items?

File: src/client/components/ConsentsSubHeader.tsx (hunk 4 above)
Comment (liamness) [reply src/client/components/ConsentsSubHeader.tsx]: Have adjusted this now.

File: src/client/components/ConsentsSubHeader.tsx (hunk 1 above)
Comment (liamness) [reply src/client/components/ConsentsSubHeader.tsx:189]:
```diff
diff --git a/src/client/components/ConsentsSubHeader.tsx b/src/client/components/ConsentsSubHeader.tsx
index 9f882d95..8909b6be 100644
--- a/src/client/components/ConsentsSubHeader.tsx
+++ b/src/client/components/ConsentsSubHeader.tsx
@@ -177,6 +177,16 @@ const screenReaderTextStyles = css`
   ${visuallyHidden}
 `;
 
+const getPageStatus = (index: number, active: number): PageStatus => {
+  if (index === active) {
+    return 'active';
+  }
+  if (index < active) {
+    return 'complete';
+  }
+  return 'pending';
+};
+
 export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
   const active = current
     ? (CONSENTS_PAGES_ARR as string[]).indexOf(current)
@@ -186,15 +196,7 @@ export const ConsentsSubHeader = ({ autoRow, title, current }: Props) => {
     <div css={[progressWrapper, autoRow()]}>
       <ol css={ol(active)}>
         {CONSENTS_PAGES_ARR.map((page, index) => {
-          const status: PageStatus = (() => {
-            if (index === active) {
-              return 'active';
-            }
-            if (index < active) {
-              return 'complete';
-            }
-            return 'pending';
-          })();
+          const status = getPageStatus(index, active);
           const screenReaderText = {
             active: 'Current: ',
             complete: 'Completed: ',
```

Something like this? I like IIFEs sometimes when I need to assign the result of something slightly complex, that can't just be a single line expression, but also isn't really used anywhere else so assigning a name to the function seems a bit redundant.

Nested ternaries could be a solution for this sort of thing, if Prettier formatted them in a way where they could be digested easily at a glance. But sadly that is not the case.

Comment (ob6160) [reply src/client/components/ConsentsSubHeader.tsx:189]: yeah it's a tricky one as there are good arguments for both sides, having it as an iife helps when I'm reading code sometimes so I'm not constantly scrolling around. i'm cool with keeping it like it is here

File: src/client/components/ConsentsSubHeader.tsx (hunk 2 above)
Comment (ob6160) [reply src/client/components/ConsentsSubHeader.tsx:198]: That makes sense, that's a really helpful doc, thanks for linking

**Comment by liamness** [comment]
[Onboarding flow redesign follow up](https://trello.com/c/ncPxVzLu/3810-onboarding-flow-redesign-follow-up)

**Pull request merged**